Update files to reflect proposed field naming convention change

BenjaminMichaelis opened the pull request:
## Summary

Detailed explanation in Issue #22822
Issue: Fields should not be named camelCase because it overlaps with local variable names
Solution: To either prefix with a _fieldName or _PropertyName so that it matches the property name.

Fixes #22822

BenjaminMichaelis committed:
Update Program.cs to reflect proposed field naming convention change

Detailed explanation in Issue #22822

Issue: Fields should not be named camelCase because it overlaps with local variable names
Solution: To either prefix with a _fieldName or _PropertyName so that it matches the property name.

BenjaminMichaelis committed:
minor edits to Program.cs

update date to _Date

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22823) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22823) before we can accept your contribution.<br/><br/>:x: BenjaminMichaelis [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22823)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=22823) it.</sub>

BenjaminMichaelis committed:
Modify public field name

Modify to match proposed naming conventions.
Detailed explanation in Issue #22822

Issue: Fields should not be named camelCase because it overlaps with local variable names
Solution: To either prefix with a _fieldName or _PropertyName so that it matches the property name.

BenjaminMichaelis committed:
Relocate private field to be right before Property

As public fields shouldn't really be used anyways, shows the naming convention of a private field relative to its property

Youssef1313 commented on docs/csharp/programming-guide/classes-and-structs/snippets/fields/Program.cs:
```diff
@@ -3,25 +3,25 @@
+    public string Day;
+    
+    // private field
+    private DateTime _Date;
```
The convention is to use `_date`.

sharwell replied:
Agreed, this is not a standard naming convention.

Youssef1313 reviewed (commented):
Have you reviewed the article(s) that references this code if they contain any changed field name?

BillWagner commented:
Thanks for starting this work @BenjaminMichaelis

As others have noted, these changes don't conform to our coding guidelines. In our docs, we intend to follow the coding standards adopted by the .NET runtime team. It's true that some of our older samples don't conform today, including this one. If you update this PR to match those guidelines, we'll review it again.

BenjaminMichaelis commented:
Switching to PR #22822
